x/go/binary/compress/compress.go

```diff
@@ -0,0 +1,173 @@
+package compress
+
+import (
+	"math"
+)
+
+// a type called Compressor which is an interace
+type Compressor interface {
+	// Compress compresses the given source bytes and outputs them
+	// to the destination bytes, returning any error encountered.
+	Compress(src []byte) (dst []byte, err error)
+}
+
+type Decompressor interface {
```

Comment:
The functions you've declared don't implement these interfaces. Should probably put them into a field less struct called `Bool`